src/model/state.ts

```diff
@@ -112,6 +115,14 @@ export class State {
     applyNewValidatorRegistered(_time: number, event: EventTypes['ValidatorRegistered']) {
         this.snapshot.CurrentIp[event.returnValues.addr] = getIpFromHex(event.returnValues.ip);
     }
+
+    applyNewImageVersion(imageName: string, imageVersion: string) {
+        if (!Versioning.isValid(imageVersion)) return;
+        const currentVersion = this.snapshot.CurrentImageVersions[imageName];
+        if (!currentVersion || Versioning.compare(imageVersion, currentVersion) > 0) {
```

Comment:
this is a major feature - to use > instaed of != this is where we enforce the no downgrade policy. although this is mostly a precaution against the paging bug (workaround) I'd add a comment explaining the reason why this is not simply !=

Reply:
good idea, there are also tests for this to make sure we can't downgrade